### DISCUSSION TITLE
Is there a way to do a reverse animation?

### DISCUSSION THREAD

**Discussion by kevinRoomies** [pmndrs/react-spring#1745] [Support]
I made a drawer component on web mobile and I'm animating it so that the grey background that block everything behind fade in first, then the drawer slides up from the bottom of the phone. But on leave, I would like to do the reverse, first the drawer sliding out and the background fading out.
The issue with chain animation is that the fading background will always be the first one to trigger. And if I use a state to tell the other animation to go on when the correct one is resting, the issue is that they both depend from the same isOpen value to start the animation on enter and leave so they're still gonna trigger together.
I'm a bit confused if that's a possible behavior.

**Comment by joshuaellis** [comment]
use the imperative api to call the animations when you want them to run, see [these docs](https://react-spring.io/common/imperatives-and-refs#imperatives--refs)

**Reply by kevinRoomies** [reply]
I'm gonna be honnest I'm struggling to grasp how that works

**Reply by joshuaellis** [reply]
If you can supply a codesandbox I can take a look 👍🏼 else we have plenty of examples on the site that use the imperative API.
